core/src/main/java/com/datastax/oss/driver/api/core/detach/AttachmentPoint.java

```diff
@@ -39,6 +40,12 @@ public CodecRegistry getCodecRegistry() {
   @NonNull
   ProtocolVersion getProtocolVersion();
 
+  /**
+   * Note that the default registry implementation returned by the driver also implements {@link
+   * MutableCodecRegistry}, which allows you to register new codecs at runtime. You can safely cast
+   * the result of this method (as long as you didn't hack the driver internals to plug a custom
```

Comment:
```
as long as you didn't hack the driver internals to plug a custom
   * registry implementation
```
maybe we can give an example of what exactly we want to prevent? Users may do this without being aware that they are doing this "hack"

Reply:
Good point, I'll be more specific like "extend the driver context to override buildCodecRegistry".